### PR TITLE
Data Analysis Label Fix for OD

### PR DESCRIPTION
**Pull request by Advitya17**
## Description

Before:

image

After:

image

image

## Checklist

- [x] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: libs/core-ui/src/lib/DatasetCohort.ts [hunk 1]
```diff
@@ -160,6 +170,13 @@ export class DatasetCohort {
             ? MulticlassClassificationEnum.Misclassified
             : MulticlassClassificationEnum.Correct;
       }
+    } else if (modelType && IsObjectDetection(modelType)) {
+      for (const [index, row] of dataDict.entries()) {
+        dataDict[index][DatasetCohortColumns.ClassificationError] =
+          row[DatasetCohortColumns.ObjectDetectionIncorrect] !== "(none)"
```
Comment (tongyu-microsoft) [inline libs/core-ui/src/lib/DatasetCohort.ts:176]: can we put this "(none)" in a constant

**Comment by imatiach-msft** [comment]
@Advitya17 Can you please improve the description?  I can't quite tell what the fix is for, and the images have no text describing whether it's before/after the fix or something else or any circle/square focusing in on what the changes were.

**Comment by Advitya17** [comment]
> @Advitya17 Can you please improve the description? I can't quite tell what the fix is for, and the images have no text describing whether it's before/after the fix or something else or any circle/square focusing in on what the changes were.

@imatiach-msft added!

File: libs/core-ui/src/lib/util/JointDataset.ts [hunk 2]
```diff
@@ -691,6 +537,246 @@ export class JointDataset {
...
+            localization.Interpret.Columns.misclassified
+          ],
+          treatAsCategorical: true
+        };
+      }
+      this.dataDict?.forEach((row, rowIndex) => {
+        row[JointDataset.ClassificationError] =
+          this.strDataDict?.[rowIndex][
+            JointDataset.ObjectDetectionIncorrect
+          ] !== "(none)"
```
Comment (imatiach-msft) [inline libs/core-ui/src/lib/util/JointDataset.ts:773]: use constant instead of str for "(none)"

**Review by imatiach-msft — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Context/buildModelAssessmentContext.ts [hunk 3]
```diff
@@ -60,6 +60,7 @@ export function buildInitialModelAssessmentContext(
     featureMetaData: props.dataset.feature_metadata,
     localExplanations,
     metadata: modelMetadata,
+    objectDetectionLabels: props.dataset.objectDetectionLabels,
```
Comment (imatiach-msft) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Context/buildModelAssessmentContext.ts:63]: This should be object_detection_labels, not objectDetectionLabels.  It looks inconsistent with all of the other fields.  We should add it to whitelist for linting like all of the other fields here, such as feature_metadata.

Comment (Advitya17) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Context/buildModelAssessmentContext.ts:63]: Yep, will merge https://github.com/microsoft/responsible-ai-toolbox/pull/2360 first which adds this support.

File: libs/core-ui/src/lib/util/JointDataset.ts [hunk 4]
```diff
@@ -691,6 +537,246 @@ export class JointDataset {
...
+        args.metadata,
+        JointDataset.PredictedYLabel,
+        localization.Interpret.ExplanationScatter.predictedY,
+        localization.Interpret.ExplanationScatter.predictedY,
+        args.targetColumn
+      );
+      this.hasPredictedY = true;
+    } else if (args.objectDetectionLabels && modelTypeIsObjectDetection) {
+      this.updateMetaDataDict(
+        args.objectDetectionLabels.map((label) => label.incorrect),
```
Comment (imatiach-msft) [inline libs/core-ui/src/lib/util/JointDataset.ts:614]: Should these be sorted?  I see inconsistencies in the order of the screenshot which is a bit confusing.

**Pull request merged**